### PR TITLE
Restrict changing direction while moving to player only.

### LINKED ISSUE #8: Support direction for object
Player has direction and can change direction when moving around
Other entities can have direction as well

### PR DESCRIPTION
**Pull request by lilydelaserre**
Restrict changing direction while moving to player only, rather than to all affected movable objects on the way.

### CODE REVIEW

File: src/systems/input_system.rs [hunk 1]
```diff
@@ -92,7 +92,8 @@ impl<'a> System<'a> for InputSystem {
                         KeyCode::Right => renderable.position.x += 1,
                         _ => ()
                     }
-                    if let Some(directional) = directionals.get_mut(entities.entity(id)) {
+                    if player.contains(entities.entity(id)) {
```
Comment (thanhqng1510) [inline src/systems/input_system.rs]: Player will definitely change direction, do not need to check if player.contains and do not need to run in the for loop

File: src/systems/input_system.rs [hunk 2]
```diff
@@ -92,7 +92,8 @@ impl<'a> System<'a> for InputSystem {
...
+                        let directional = directionals.get_mut(entities.entity(id)).unwrap();
                         match key {
                             KeyCode::Up => directional.direction = Direction::Up,
```
Comment (thanhqng1510) [inline src/systems/input_system.rs]: Many duplicates code, should be 
```suggestion
                            directional.direction = match key {
                                    KeyCode::Up => Direction::Up,
                            }
```

File: src/systems/input_system.rs (hunk 1 above)
Comment (thanhqng1510) [inline src/systems/input_system.rs]: Currently only change the player direction if the player can move, however the player should be able to change direction even when in stuck

**Review by thanhqng1510 — CHANGES_REQUESTED** [review]
Wrong logic, need to be fixed

**Comment by lilydelaserre** [comment]
> Wrong logic, need to be fixed
> Many duplicates code, should be

Logic fixed and code shortened.

**Pull request merged**